tiledb/sm/c_api/tiledb_experimental.h

```diff
@@ -171,6 +171,454 @@ TILEDB_EXPORT int32_t tiledb_array_evolve(
+    tiledb_ctx_t* ctx,
+    const tiledb_subarray_t* subarray,
+    tiledb_subarray_partitioner_t** subarray_partitioner,
+    uint64_t memory_budget,
+    uint64_t memory_budget_var,
+    uint64_t memory_budget_validity);
+
+/**
+ * Frees a TileDB subarray partitioner object.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_t* subarray;
+ * tiledb_subarray_partitioner_t *subarray_partitioner;
+ * tiledb_subarray_partitioner_alloc(ctx, subarray, &subarray_partitioner);
+ * tiledb_subarray_partitioner_free(&subarray_partitioner);
+ * @endcode
+ *
+ * @param subarray_partitioner The subarray partitioner object to be freed.
+ */
+TILEDB_EXPORT void tiledb_subarray_partitioner_free(
+    tiledb_subarray_partitioner_t** subarray_partitioner);
+
+/**
+ * Set the layout of the subarray associated with a subarray partitioner object.
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_t* subarray;
+ * tiledb_subarray_partitioner_t *subarray_partitioner;
+ * tiledb_subarray_partitioner_set_layout(ctx, layout, subarray_partitioner);
+ * @endcode
+ *
+ * @param layout The layout to set into the subarray partioner object's
+ * subarray.
+ *    - `TILEDB_COL_MAJOR`:
+ *      This means column-major order with respect to the subarray.
+ *    - `TILEDB_ROW_MAJOR`:
+ *      This means row-major order with respect to the subarray.
+ *    - `TILEDB_GLOBAL_ORDER`:
+ *      This means that cells are stored or retrieved in the array global
+ *      cell order.
+ *    - `TILEDB_UNORDERED`:
+ *      This is applicable only to reads and writes for sparse arrays, or for
+ *      sparse writes to dense arrays. For writes, it specifies that the cells
+ *      are unordered and, hence, TileDB must sort the cells in the global cell
+ *      order prior to writing. For reads, TileDB will return the cells without
+ *      any particular order, which will often lead to better performance.
+ * * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_set_layout(
+    tiledb_ctx_t* ctx,
+    tiledb_layout_t layout,
+    tiledb_subarray_partitioner_t* partitioner);
+
+/**
+ * Compute the complete series of partitions/subarrays leaving them stored
+ * internally in the object.
+ *
+ * These partitions can then be retrieved via
+ * tiledb_subarray_partitioner_get_partition().
+ *
+ * **Example:**
+ *
+ * @code{.c}
+ * tiledb_subarray_partitioner_t *subarray_partitioner;
+ * tiledb_subarray_partitioner_compute(ctx, subarray_partitioner);
+ * @endcode
+ *
+ * @param The partitioner for which to compute the complete series of (sub)
+ * partitions
+ * @return `TILEDB_OK` for success or `TILEDB_ERR` for error.
+ */
+TILEDB_EXPORT int32_t tiledb_subarray_partitioner_compute(
+    tiledb_ctx_t* ctx, tiledb_subarray_partitioner_t* partitioner);
+
+/**
+ * Gets number of computed partitions available to be retrieved via
+ * tiledb_subarray_partitioner_get_partition().
```

Comment:
tiledb_subarray_partitioner_get_partition_num

Reply:
Not sure of your intent, this comment applies to api tiledb_subarray_partitioner_get_partition_num() and is informing that same is returning the number of partitions that can be retrieved (after they are computed) via tiledb_subarray_partitioner_get_partition().